tests/scripts/depends.py

```diff
@@ -140,22 +140,22 @@ def option_exists(conf, option):
         return False
     return True
 
-def set_config_option(conf, option, colors, value=None):
-    """Set configuration option, optionally specifying a value"""
+def set_config_option_value(conf, option, colors, value):
+    """Set/unset a configuration option, optionally specifying a value"""
     if not option_exists(conf, option):
         log_line('Symbol {} was not found in {}'.format(option, conf.filename), color=colors.red)
         return False
-    log_command(['config.py', 'set', option])
-    conf.set(option, value)
-    return True
 
-def unset_config_option(conf, option, colors):
-    """Unset configuration option if it exists"""
-    if not option_exists(conf, option):
-        log_line('Symbol {} was not found in {}'.format(option, conf.filename), color=colors.red)
-        return False
-    log_command(['config.py', 'unset', option])
-    conf.unset(option)
+    if value is False:
+        log_command(['config.py', 'unset', option])
+        conf.unset(option)
+    else:
+        if value is True:
```

Comment:
I assume that `value` can be of `Boolean` or `String` type?
Not sure if this is good solution. Safer would be to keep `value` always as `String` and assume possible values like: `Enable`, `Disable`, `<Other Value>`.

Then we would have:
`set_config_option_value(conf, 'MBEDTLS_TEST_HOOKS', colors, 'Disable')`

Reply:
We need to distinguish `#define FOO` from `#define FOO Enable`, so no, the current code is right in having a string for `#define FOO ...` vs `True` for `#define FOO` and `False` for no definition at all.

It would help to convey this through type annotations. `value : Union[bool, str]` and documentation explaining what I wrote above.

Reply:
Added as a new commit.